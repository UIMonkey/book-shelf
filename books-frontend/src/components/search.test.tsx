import { render } from '@testing-library/react';
import Search, { ISearchComponent } from './search';


describe('Search component tests', () => {
    let searchInput: ISearchComponent;
    let searchComponent: JSX.Element;

    beforeAll(() => {
        searchInput = {
            value: 'timmy',
            children: 'john',
            onChange: () => { }
        }
    });

    beforeEach(() => {
        searchComponent = <Search value={searchInput.value} children={searchInput.children} onChange={searchInput.onChange} />
    });

    test('It renders the Search component', () => {
        render((searchComponent));
    });

    test('It matches the search snapshot', () => {
        const { container } = render(searchComponent);
        expect(container.firstChild).toMatchSnapshot();
    });
});