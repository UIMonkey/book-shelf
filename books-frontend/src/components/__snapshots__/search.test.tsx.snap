// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component tests It matches the search snapshot 1`] = `
<div>
  <label
    for="search"
  >
    john
  </label>
  <input
    id="search"
    type="text"
    value="timmy"
  />
</div>
`;
